### PR TITLE
[AKS] Fix #22032: `az aks nodepool add/update`: Fix autoscaler parameters for user node pools

### LINKED ISSUE #22032: az aks nodepool update --min-count cannot be 0 after az cli update
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.62

**Describe the bug**
After cli update to version 2.35.0 from 2.33.1 we started getting `--min-count must be in the range [1,100]` errors when running 
this command `az aks nodepool update --name <nodepool> --cluster-name <cluster>  --resource-group <rg> --min-count 0 --max-count 2 --update-cluster-autoscaler --mode user`
Setting nodepool min-count to 0 is possible both from older version of cli and azure portal. Was this change intentional?

**To Reproduce**
Run `aks nodepool update --name <nodepool> --cluster-name <cluster>  --resource-group <rg> --min-count 0 --max-count 2 --update-cluster-autoscaler --mode user` on user nodepool with autoscaler enabled

**Expected behavior**
Nodepool should be updated instead of throwing validation errors

**Environment summary**
CLI version: 2.35.0

**Additional context**

### LINKED ISSUE #22438: az aks nodepool add --max-count range should be 1, 1000
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.62

**Related command**

az aks nodepool add ...  --max-count 1000

**Describe the bug**

az aks nodepool add ...  --max-count 1000  is not allowed due to the validation check (range must be in [1,100]). 
Based on the documentation (https://docs.microsoft.com/en-us/cli/azure/aks/nodepool?view=azure-cli-latest#az-aks-nodepool-add-optional-parameters), it should be range up to 1000. 
**To Reproduce**

az aks nodepool add ... --enable-cluster-autoscaler --max-count 1000 
**Expected behavior**

It should be able to take 1000 as the max-count.
**Environment summary**

**Additional context**

### PR DESCRIPTION
**Pull request by PramodValavala-MSFT**
**Related command**

```
az aks nodepool add
az aks nodepool update
```

**Description**

This PR address 2 changes

- Node Pools can be scaled from 0 to 1000 nodes from the Azure Portal, but the CLI has a check restricting them between 1 and 100
- Only User Node Pools support scaling to 0 nodes, which after the fix for the acceptable range above, results in unintuitive error messages from the ARM endpoint

After the fix for the first, the following commands were tested, resulting in errors from ARM as below
```sh
# Creating a System Node Pool with Min Count set to 0 and Node Count set to 0
$ az aks nodepool add --name {} --cluster-name {} --resource-group {} --enable-cluster-autoscaler --node-count 0 --min-count 0 --max-count 1 --mode System
(InvalidParameter) The value of parameter agentPoolProfile.count is invalid. Please see https://aka.ms/aks-naming-rules for more details.
Code: InvalidParameter
Message: The value of parameter agentPoolProfile.count is invalid. Please see https://aka.ms/aks-naming-rules for more details.
Target: agentPoolProfile.count

# Creating a System Node Pool with Min Count set to 0 and Node Count set to 1
$ az aks nodepool add --name {} --cluster-name {} --resource-group {} --enable-cluster-autoscaler --node-count 1 --min-count 0 --max-count 1 --mode System
(InvalidParameter) The value of parameter agentPoolProfile.minCount is invalid. Please see https://aka.ms/aks-naming-rules for more details.
Code: InvalidParameter
Message: The value of parameter agentPoolProfile.minCount is invalid. Please see https://aka.ms/aks-naming-rules for more details.
Target: agentPoolProfile.minCount

# Updating a System Node Pool to have Min Count set to 0
$ az aks nodepool update --name {} --cluster-name {} --resource-group {} --update-cluster-autoscaler --min-count 0 --max-count 1
(InvalidParameter) The value of parameter agentPoolProfile.minCount is invalid. Please see https://aka.ms/aks-naming-rules for more details.
Code: InvalidParameter
Message: The value of parameter agentPoolProfile.minCount is invalid. Please see https://aka.ms/aks-naming-rules for more details.
Target: agentPoolProfile.minCount

# Updating a User Node Pool to a System Node Pool with Min Count set to 0
$ az aks nodepool update --name {} --cluster-name {} --resource-group {} --update-cluster-autoscaler --min-count 0 --max-count 1 --mode System
(InvalidParameter) The value of parameter agentPoolProfile.minCount is invalid. Please see https://aka.ms/aks-naming-rules for more details.
Code: InvalidParameter
Message: The value of parameter agentPoolProfile.minCount is invalid. Please see https://aka.ms/aks-naming-rules for more details.
Target: agentPoolProfile.minCount
```

With the other set of changes in this PR, all of the above commands will result in the following error instead
```
Value of min-count should be greater than or equal to 1 for System node pools
```

resolves #22032
resolves #22438

**Testing Guide**

Run the commands mentioned above

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/azure-cli/azure/cli/command_modules/acs/_help.py
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/_help.py:988]: User could add a system mode nodepool, the value 0 would be invalid for this case.

File: src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py:652]:
```suggestion
            mode=self.get_mode(),
```

File: src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py:643]:
```suggestion
```

File: src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py:699]:
```suggestion
```

File: src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py:722]:
```suggestion
            mode=self.get_mode(),
```

File: src/azure-cli/azure/cli/command_modules/acs/_help.py
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/_help.py:988]: May add something like `If it is a system nodepool, the valid minimum value is 1.`

Comment (andyliuliming) [reply src/azure-cli/azure/cli/command_modules/acs/_help.py:988]: Minimum nodes count used for autoscaler, when "--enable-cluster-autoscaler" specified. Please specify the value in the range of [0, 1000] for user nodepool, and [1,1000] for system nodepool.

Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/_help.py:988]:
```suggestion
    short-summary: Minimum nodes count used for autoscaler, when "--enable-cluster-autoscaler" specified. Please specify the value in the range of [0, 1000] for user nodepool, and [1,1000] for system nodepool.
```

File: src/azure-cli/azure/cli/command_modules/acs/_help.py
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/_help.py:991]:
```suggestion
    short-summary: Minimum nodes count used for autoscaler, when "--enable-cluster-autoscaler" specified. Please specify the value in the range of [0, 1000] for user nodepool, and [1,1000] for system nodepool.
```

File: src/azure-cli/azure/cli/command_modules/acs/_help.py
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/_help.py:1104]:
```suggestion
    short-summary: Minimum nodes count used for autoscaler, when "--enable-cluster-autoscaler" specified. Please specify the value in the range of [0, 1000] for user nodepool, and [1,1000] for system nodepool.
```

File: src/azure-cli/azure/cli/command_modules/acs/_help.py
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/_help.py:1107]:
```suggestion
    short-summary: Minimum nodes count used for autoscaler, when "--enable-cluster-autoscaler" specified. Please specify the value in the range of [0, 1000] for user nodepool, and [1,1000] for system nodepool.
```

**Comment by andyliuliming** [comment]
will approve this after @FumingZhang add one more live test case.

**Comment by FumingZhang** [comment]
Here's the [PR #22451](https://github.com/Azure/azure-cli/pull/22451) to add live test cases. Will update recordings after this PR is merged.

**Review by andyliuliming — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**
